packages/core/src/actions/assign.ts

```diff
@@ -146,7 +147,16 @@ export function assign<
         TEvent,
         TActor
       >
-): AssignAction<TContext, TExpressionEvent, TParams, TEvent, TActor> {
+): ActionFunction<
```

Comment:
We could revert this if we'd figure out how to include `| undefined` in our constraints for contextual type parameters (`_out_TActor` etc)